backend/src/controllers/items-controller.ts

```diff
@@ -0,0 +1,136 @@
+    }
+    //Duplicated item check
+    const dupeItem = Item.findOne({name: req.body.name})
+    if (!dupeItem) {
+        return res.status(409).json({error: "An item with the same name already exists."});
+    }
+    //Checks if the ID matches the owner of the item
+    const itemOwnerId = req.body._id;
+    const itemOwnerExistence = await User.findOne({_id: itemOwnerId});
+    if (!itemOwnerExistence) {
+        return res.status(404).json({error: "This id does not match the owner of the item."});
+    }
+    const name = req.body.name as string;
+    const description = req.body.description as string;
+    const pickupLocation = req.body.pickupLocation as string;
+    const pickupTime = req.body.pickupTime as string;
+
+
+
+    //Interface for a new item
+    const item = new Item({
+        name: name,
+        description: description,
+        pickupLocation: pickupLocation,
+        pickupTime: pickupTime,
+        itemOwnerId: itemOwnerExistence._id
+    });
+    //Creates a transaction to create inventory-items
+    //Checks if there's missing sizes
+    const transaction = await mongoose.startSession();
+    transaction.startTransaction();
+    try {
+        const savedItem = await item.save();
+        const sizeList = ['XXS', 'XS', 'S', 'M', 'L', 'XL', 'XXL'];
+        for (const size of sizeList) {
+            const invItem = new InventoryItem({
+                size: size,
+                item: savedItem._id
+            });
+            await invItem.save({session: transaction});
+        }
+        await transaction.commitTransaction();
+        await transaction.endSession();
+        return res.status(201).json({item});
+    } catch (error: any) {
+        await transaction.abortTransaction();
+        await transaction.endSession();
+        return res.status(500).json({message: 'Server error', error: error.message});
+    }
+}
+
+
+//Retrieves a list of all items
+export const retrieveItems = async (req: Request, res: Response) => {
+    try {
+        const items = await Item.find();
+        return res.status(200).json({items});
+    } catch (error: any) {
+        return res.status(500).json({ message: 'Server error', error: error.message });
+    }
+}
+
+//Retrieves details of a specific item
+export const getSpecificItem = async (req: Request, res: Response) => {
+    try {
+        const itemID: string = req.params._id;
+        const desiredItem = await Item.findOne({ _id: itemID });
+        if (desiredItem){
+            return res.status(200).json({item: desiredItem});
+        }
+        else{
+            return res.status(404).json({ error: "No item with the provided ID exists"});
+        }
+    } catch (error: any){
+        return res.status(500).json({ message: 'Server error', error: error.message });
+    }
+}
+
+//Updates a specific item
+export const updateItem = async (req: Request, res: Response) => {
+    const { newName }: { newName: string }= req.body;
```

Comment:
When I call to this route --

![image](https://github.com/miguelaenlle/merch-order-request-form/assets/66172204/8c961664-fa7d-4dba-9b9b-355c13c54e16)

There's a mismatch between the requested updates and the actual updates.

Reply:
Check the fields in routes -- they don't all line up with what's retrieved in the function.